File: Anya_Seth/Sliding Window/Subarray Product Less Than K.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
      if(k <= 1){
        return 0;
      }
      int prod = 1;
      int count = 0;
      int start = 0;
      int end = 0;
      while(end < nums.length){
        prod = prod * nums[end];
        while(prod >= k){
            prod = prod / nums[start];
            start++;
        }
        count += count + (end - start + 1);
        end++;
      } 
      return count;
    }
}

File: NikhilKumarGupta/Slidingwindow/Subarray Product Less Than K.cpp
class Solution {
    public:
        int numSubarrayProductLessThanK(vector<int>& nums, int k) {
            int start=0;
            int n=nums.size();
            int count=0;
            int product=1;
            if(k<=1)return 0;
    
            for(int end=0;end<n;end++){
                product*=nums[end];
    
    
                while(product>=k){
                    product/=nums[start];
                    start++;
                }
              
                count+=(end-start+1);
                
    
            }
            return count;
            
        }
    };